test/org/zaproxy/zap/extension/pscanrules/TestInfoPrivateAddressDisclosureUnitTest.java

```diff
@@ -0,0 +1,286 @@
+		return new TestInfoPrivateAddressDisclosure();
+	}
+
+	private HttpMessage createHttpMessage(String body) throws HttpMalformedHeaderException {
+		HttpMessage msg = new HttpMessage();
+		msg.setRequestHeader("GET https://www.example.com/ HTTP/1.1");
+		msg.setResponseHeader("HTTP/1.1 200 OK\r\n");
+		msg.setResponseBody(body);
+		return msg;
+	}
+
+
+	@Test
+	public void alertsIfPrivateIp() throws HttpMalformedHeaderException {
+		String[][] data = new String[][] {
+				// IPs defined in RFC 1918
+				{ "10.0.0.0",           "10.0.0.0" },
+				{ "10.10.10.10",        "10.10.10.10" },
+				{ "10.255.255.255",     "10.255.255.255" },
+				{ "172.16.0.0",         "172.16.0.0" },
+				{ "172.25.16.32",       "172.25.16.32" },
+				{ "172.31.255.255",     "172.31.255.255" },
+				{ "192.168.0.0",        "192.168.0.0" },
+				{ "192.168.36.127",     "192.168.36.127" },
+				{ "192.168.255.255",    "192.168.255.255" },
+				// some OK stuff (?)
+				{ "10.0.0.0:",          "10.0.0.0" },
+				{ "10.0.0.0:6553",      "10.0.0.0:6553" },
+				{ " 10.0.0.0 ",         "10.0.0.0" },
+				{ "/10.0.0.0.",         "10.0.0.0" },
+				{ ";10.0.0.0,",         "10.0.0.0" },
+				{ "\n10.0.0.0\t",       "10.0.0.0" },
+				{ "\n10.0.0.0\t",       "10.0.0.0" },
+				// desired functionality?
+				{ "10.0.0.0:bla",       "10.0.0.0:" },
+				{ "15.10.0.0.0.12.27",  "10.0.0.0" },
+				{ "255.10.0.0.0:6555",  "10.0.0.0:6555" },
+				{ "100.10.0.0.0.10.12", "10.0.0.0" }, 
+				{ "2050:10.0.0.0bla",   "10.0.0.0" },
+				{ "205010.0.0.0bla",    "10.0.0.0" },
+				{ "abcd10.0.0.0bla",    "10.0.0.0" },
+				{ "abcd10.0.0.999",     "10.0.0.99" },
+				{ "abcd10.0.0.9999999", "10.0.0.99" },
+				{ "ip-10.0.0.0",        "10.0.0.0" }
+		};
+		for (int i = 0; i < data.length; i++) {
+			String candidate = data[i][0];
+			String evidence = data[i][1];
+			HttpMessage msg = createHttpMessage(candidate);
+			rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+
+			assertThat(alertsRaised.size(), equalTo(i + 1));
+			assertThat(alertsRaised.get(i).getEvidence(), equalTo(evidence));
+		}
+	}
+
+	@Test
+	public void passesIfNonPrivateOrInvalidIp() throws HttpMalformedHeaderException {
+		String[] candidates = new String[] {
+				// define the "borders" of private IP ranges
+				"9.255.255.255",
+				"11.0.0.0",
+				"172.15.255.255",
+				"172.32.0.0",
+				"192.167.255.255",
+				"192.169.0.0",
+				// somewhere outside & between the private ranges
+				"8.8.8.8",
+				"26.10.3.11",
+				"84.168.27.12",
+				"127.0.0.1",	// the original comment said: "IP's including localhost" should we add it?
+				"186.27.16.2",
+				"255.255.255.255",
+				// some invalid ones
+				"10",
+				"10.0.0",
+				"10.0.0:0",
+				"10/0.0.0",
+				"10.0\n0.0",
+				"10.0.0 0",
+				"10.0.0.a",
+				"999.0.0.0", 
+				"10.999.0.0",
+				"10.0.756.0",
+				// desired functionality?
+//				"10.0.0.999", // would raise an alert, pls. find the others in privateIps()
```

Comment:
Yepp, see line 75: `{ "abcd10.0.0.999", "10.0.0.99" }`, the last digit is handled as "surrounding" text, like the `abcd`